src/main/java/Ninon/Command.java

```diff
@@ -0,0 +1,24 @@
+package Ninon;
+
+/**
+ * Represents an abstract command that can be executed on a task list.
+ * Subclasses should implement specific behaviors for different types of commands.
+ */
+public abstract class Command {
+
+    /**
+     * Executes the command with the given task list, UI, and storage.
+     *
+     * @param taskList The task list to modify.
+     * @param ui The user interface to interact with the user.
+     * @param storage The storage to handle saving and loading of tasks.
+     */
+    public abstract void execute(TaskList taskList, Ui ui, Storage storage);
+
+    /**
+     * Determines if the command signals an exit from the program.
+     *
+     * @return true if the command is an exit command, false otherwise.
+     */
+    public abstract boolean isExit();
```

Comment:
Does this need to be abstract since you only have 1 exit command (which can override this method)? 
```suggestion
public boolean isExit() {
   return false;
}
```